### PR TITLE
core branch descriptions

### PR DESCRIPTION
**Pull request by nitzmahone**
##### SUMMARY
* documents the new `milestone` branch
* documents other core branches, branch points, policies

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

### CODE REVIEW

File: docs/docsite/rst/community/development_process.rst
Comment (dmsimard) [inline docs/docsite/rst/community/development_process.rst:39]: Is the implication here that backport PRs should not be proposed by contributors ?

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (dmsimard) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: Would you expect downstream distribution packaging to rely on GitHub instead of PyPI for the source tarball, then ?

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (gundalow) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: Will new `roadmap` files include dates for development phases and a list of what will be included in each phase?
(as well as a backlink to this file)?

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (sivel) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: I suppose that depends on the definition of downstream, but I see no problems with that.  We're still figuring things out, but there will come a time where an upstream gets shorter support than "downstream" (Red Hat) and we will no longer publish releases to PyPI, however downstream RH builds will continue, using GH for the source.

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (sivel) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: The roadmap already includes this in my PR, however the dates haven't been approved as of yet: https://github.com/ansible/ansible/pull/75843/files

File: docs/docsite/rst/community/development_process.rst
Comment (sivel) [reply docs/docsite/rst/community/development_process.rst:39]: I think it means that generally speaking, all PRs should be made against the devel branch.  Instructions will be given about whether backports are needed once the devel PR is merged.  Maybe it could use more clarification, but the core team generally does not create backports for contributors.

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (sivel) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: I tweaked this date slightly in the merged PR.  Beta 1 happened Sept 27, but I didn't have Dev Phase 1 and Beta 1 linked.

```suggestion
* 27-Sep-2021: 2.13 Development Phase 1 begins; ``milestone`` contents are updated to 2.12.0b1 with version number set to
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
tarball of the tag contents is considered the official canonical release artifact.
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
``ansible-core`` project branches and tags
```

Comment (samccann) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: We're still inconsistent in the docs, but we are supposed to use 'Sentence case' for headings, meaning only the first word is capitalized (though not in this instance since it's the package name).

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
``devel`` branch
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
``stable-X.Y`` branches
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
``vX.Y.Z`` tags
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: more of a side comment  - when we do get to that point (RH keeps going but pypi stops) - we'll need to consider what we do for docs. Porting guide specifically comes to mind, and the changelogs - if the pypi stopped at say X.Y.5, but RH keeps going for another handful of dot releases, do we keep updating the porting guide/changelogs? do we stop? Do we put some kind of special marker to say X.Y.6 release and beyond are only available through RH subscription?  Just things to consider when the time comes.

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
``milestone`` branch
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
features under active development. As described in the `roadmap <https://docs.ansible.com/ansible-core/devel/roadmap/ansible_core_roadmap_index.html>`_  for a given release, development is
```

Comment (samccann) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: Another oddity - the old roadmaps is no longer used. Core and Ansible have different roadmaps. So pointing to the core roadmap directly here. I still need to figure out why a ref to the core roadmap doesn't work in the Ansible 5 etc package docs.

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
to the potential instability of the daily commit "fire hose" from ``devel``. When a release reaches the Beta 1 milestone,
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
The following example is for illustrative purposes only. Please check the most recent roadmap for accurate dates. For example, the ``milestone`` branch in 2.13 ``ansible-core`` roadmap is scheduled to update as follows:
```

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]:
```suggestion
```

Comment (samccann) [reply docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: Suggesting these changes so we don't have to maintain roadmap dates in 2 places for 2.13...

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (samccann) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst:1]: If you choose NOT to move all of this into the development_process.rst, then please add `:orphan:` to the top of this page to prevent docs build warnings.

File: docs/docsite/rst/dev_guide/core_branches_and_tags.rst
Comment (gundalow) [inline docs/docsite/rst/dev_guide/core_branches_and_tags.rst]: Should these dates move to roadmap?

**Comment by samccann** [comment]
@nitzmahone hey there! This one has kind of lost its 'umph' since we are beyond all the branch dates, but I wanted to verify we will continue with milestone branches in future releases before seeing what needs to happen here to get this merged...

**Comment by samccann** [comment]
Adding some comments about milestone branching that happened in matrix/irc recently. Attempting a summary:

- Some users of the milestone branches might be assuming they are mini 'stable' points in core development. They are not. They are just 'points in time' where a user can decide to move to a new milestone branch in their CI and be ready to address any problems they see. These milestone branches are still devel and still subject to the usual potential breakages of a work-in-progress development branch. So rather than pull from devel and have breaking problems potentially show up at any time, the user can take a milestone branch and know at that point, have staff ready to address any breaking problems. Then stay on that milestone until the next one is released (giving them a few weeks of time before they have to address any new breaking problems)
- These are not like using 'stable-2.13' the moment it is created, but before release. The expectation is that people wouldn't add a new stable branch to their CI until RC, unless they specifically do want to test out and report problems to core developers to help drive the new stable branch to release ready.

^^ are just my notes and may not reflect the exact purpose, but I didn't want to lose the conversation.

**Comment by samccann** [comment]
@nitzmahone I'm going to take this PR over and see if I can get it merged in the next week or so.

**Review by samccann — APPROVED** [review]

**Pull request merged**
